File: NERP.Admin.Web1/ClientApp/src/app/entities/components/supplier.ts
import { Component, Input } from '@angular/core';
import { Supplier } from '../../common/models/supplier';

@Component({
    selector: 'suppliers',
    templateUrl: './supplier.html'
})
export class SupplierComponent {

    @Input() public supplierList: Supplier[] = new Array <Supplier>();

    constructor() { }

    addNewSupplier() {

        if (this.supplierList == null)
            this.supplierList = [];

        this.supplierList.push({
            name: "",
            sku: "",
            price: null,
            validFrom: "",
            validTo: ""
        });
    }

    deleteSupplier(index: number) {
        this.supplierList.splice(index, 1);
    }

}

File: NERP.Common/Models/Supplier.cs
using System;

namespace NERP.Common.Models
{
    public class Supplier
    {
        public string Name { get; set; }
        public string Sku { get; set; }
        public double? StockQty { get; set; }
        public double? Price { get; set; }
        public string Currency { get; set; }
        public DateTime? ValidFrom { get; set; }
        public DateTime? ValidTo { get; set; }
        public bool IsPrefered { get; set; }
        public double? ServiceCost { get; set; }
        public string Default { get; set; }
    }
}
